### PR TITLE
Requests

### PR DESCRIPTION
**Pull request by galenheuer**
created a table to show requests for admin to approve, working on implementing checkboxes

### CODE REVIEW

File: client/src/components/AdminTable.jsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    return {
+      checked, name, username, email, role, dateJoined: desiredDate, approved, id,
+    };
+  }
+
+  useEffect(() => {
+    setTable(users.map((user) => createData(false, user.name, user.username, user.email, user.role, user.epochDate, user.status, user.id)));
+  }, [users]);
+
+  // setTable(rows);
```
Comment (erinkwon01) [inline client/src/components/AdminTable.jsx:37]: Lines 37, 38 are those comments necessary?

File: client/src/components/AdminTable.jsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+  // }, [rows]);
```
Comment (erinkwon01) [inline client/src/components/AdminTable.jsx:38]: Are these comments necessary?

File: client/src/components/AdminTable.jsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+      const selectedAccounts = [];
+      if (deleteButton) {
+        table.forEach((user) => {
+          if (user.checked) {
+            selectedAccounts.push(user.id);
+          }
+        });
+
+        api.batchDeleteProfile(selectedAccounts);
+        // setTimeout(() => { window.location.reload(); }, 800);
```
Comment (erinkwon01) [inline client/src/components/AdminTable.jsx:88]: Is this comment necessary?

File: client/src/components/AdminTable.jsx [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+        setDeleteButton(false);
+      }
+    }
+
+    DeleteButtonHandler();
+  }, [deleteButton]);
+
+  // useEffect(() => console.log(table), [table]);
```
Comment (erinkwon01) [inline client/src/components/AdminTable.jsx:96]: Is this comment necessary?

File: client/src/components/AdminTable.jsx [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+  function handleChange(e, username) {
+    setTable((prevValue) => prevValue.map((user) => {
+      // console.log(prevValue.checked);
```
Comment (erinkwon01) [inline client/src/components/AdminTable.jsx:100]: I think you can remove this!

**Review by danielkao14 — CHANGES_REQUESTED** [review]
A slight bug:
- pressing delete doesn't refresh the page automatically (functionality seems to work)

File: client/src/pages/Requests.jsx [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+    if (value === 0) {
+      setCurrentAccounts(pendingUsers);
+    } else if (value === 1) {
+      setCurrentAccounts(allUsers);
+    }
+  }, [value]);
+
+  useEffect(() => {
+    async function fetchProfiles() {
+      // ask jerry how to get the data sorted
```
Comment (erinkwon01) [inline client/src/pages/Requests.jsx:73]: Can this comment be deleted?

File: client/src/pages/Requests.jsx [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+      <TabPanel value={value} index={1}>
+        {currentAccounts
+            && <AdminTable users={currentAccounts} setRowsSelected={setNumChecked} cancelButton={cancelButton} setCancelButton={setCancelButton} approveButton={approveButton} setApproveButton={setApproveButton} deleteButton={deleteButton} setDeleteButton={setDeleteButton} />}
+      </TabPanel>
+      <p>
+        {numChecked}
+        {' '}
+        rows are selected
+      </p>
+      {/* <button type="button">Submit</button> */}
```
Comment (erinkwon01) [inline client/src/pages/Requests.jsx:138]: Can this comment be deleted?

File: client/src/pages/Requests.jsx [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+      <button type="button" onClick={HandleCancelClick}>Cancel</button>
+      <button type="button" onClick={HandleApproveClick}>Approve</button>
+      <button type="button" onClick={HandleDeleteClick}>Delete</button>
+      {/* <p>{numApproved}</p> */}
```
Comment (erinkwon01) [inline client/src/pages/Requests.jsx:142]: Can this comment be deleted?

File: client/src/pages/UserNotApproved.jsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
+/* eslint-disable no-unused-vars */
```
Comment (erinkwon01) [inline client/src/pages/UserNotApproved.jsx:1]: Can you delete this comment?

**Review by erinkwon01 — COMMENTED** [review]
Heyy great job overall, this feature is super rad! The error that I found was that, the waiting message _does_ indeed show after I sign up for a new account, BUT I can immediately log in after going back to the Login page. I think the error I got as about updateAppProfile being undefined!

**Review by xu-jerry — APPROVED** [review]
Looks good!

**Pull request merged**